packages/headless/src/features/insight-search/insight-search-request.ts

```diff
@@ -25,10 +25,12 @@ export const buildInsightBaseRequest = async (
   return mapSearchRequest<InsightQueryRequest>({
     accessToken: state.configuration.accessToken,
     organizationId: state.configuration.organizationId,
-    url: getOrganizationEndpoint(
-      state.configuration.organizationId,
-      state.configuration.environment
-    ),
+    url:
+      state.configuration.search.apiBaseUrl ??
```

Comment:
One small issue I see with using the `search.apiBaseUrl` value from the configuration state here is that it makes it difficult to document the property correctly (right now it is https://github.com/coveo/ui-kit/pull/4339/files#diff-a4f8bb3532fc4170d823e08a8962590441a17be552d7c73f897e3e3c19ad9b5eR52-R62, but this would no longer be accurate when seen in the context of the configuration state of an insight engine).

Eventually, we'll make the engine internal, so it won't really matter anymore, but until then we should at least ensure that the documentation correctly reflects what the property does.

An alternative would be to add an insight-specific configuration section, but that's a way more involved change, so I wouldn't necessarily do that. Fixing the docs seems like the simplest / best solution for now.

Reply:
Indeed we need to make the doc generic enough for the insight and search use cases